src/api/app/controllers/source_controller.rb

```diff
@@ -167,27 +206,27 @@ def render_project_verboseproductlist
   # DELETE /source/:project
   def delete_project
     project_name = params[:project]
-    pro = Project.get_by_name project_name
+    project = Project.get_by_name project_name
 
     # checks
-    unless User.current.can_modify_project?(pro)
+    unless User.current.can_modify_project?(project)
       logger.debug "No permission to delete project #{project_name}"
       render_error :status => 403, :errorcode => 'delete_project_no_permission',
                    :message => "Permission denied (delete project #{project_name})"
       return
     end
-    pro.can_be_deleted?
+    project.can_be_deleted?
 
     # find linking repos
-    private_check_and_remove_repositories(params, pro.repositories) or return
+    project.check_and_remove_repositories(params, project.repositories) or return
```

Comment:
It looks like check_and_remove_repositories always returns true and otherwise raises an error. So as for right now the 'or return' isn't needed